packages/react/src/components/TopNavigation/TopNavigation.tsx

```diff
@@ -0,0 +1,231 @@
+   */
+  onLeftNavigationTrigger?: () => void;
+  /**
+   * Log out function.
+   */
+  onLogOut?: () => void;
+  /**
+   * List of themes.
+   */
+  themes?: ThemeListInterface[];
+  /**
+   * Logged user information.
+   */
+  user?: UserTemplate;
+}
+
+/**
+ * Interface for the brand template.
+ */
+export interface BrandTemplate {
+  logo?: {
+    desktop?: ReactNode;
+    mobile?: ReactNode;
+  };
+  onClick?: () => void;
+  title?: ReactNode;
+}
+
+/**
+ * Interface for the logged user template.
+ */
+export interface UserTemplate {
+  email?: string;
+  image?: string;
+  name?: string;
+}
+
+const COMPONENT_NAME: string = 'TopNavigation';
+
+/**
+ * Top Navigation component.
+ */
+const TopNavigation: FC<TopNavigationProps> & WithWrapperProps = (props: TopNavigationProps): ReactElement => {
+  const {
+    className,
+    children,
+    isLeftNavigationActive,
+    brand,
+    user,
+    links,
+    themes,
+    onLogOut,
+    onLeftNavigationTrigger,
+    ...rest
+  } = props;
+
+  const [anchorElUser, setAnchorElUser] = useState<null | HTMLElement>(null);
+  const [selectedTheme, setSelectedTheme] = useState('System Default');
+
+  const theme: Theme = useTheme();
+  const isMobile: boolean = useMediaQuery(theme.breakpoints.down('sm'));
+  const classes: string = clsx(
+    'oxygen-top-nav',
+    {
+      mobile: isMobile,
+    },
+    className,
+  );
+
+  const open: boolean = Boolean(anchorElUser);
+
+  const handleOpenUserMenu = (event: MouseEvent<HTMLElement>): void => {
+    setAnchorElUser(event.currentTarget);
+  };
+
+  const handleCloseUserMenu = (): void => {
+    setAnchorElUser(null);
+  };
+
+  // TODO: take default themes from `Theme`.
+  const defaultThemes: ThemeListInterface[] = [
+    {icon: <EclipseIcon />, name: 'System Default'},
+    {icon: <SunIcon />, name: 'Light'},
+    {icon: <CresentBrightIcon />, name: 'Dark'},
+    {icon: <SunContrastIcon />, name: 'High Contrast'},
+  ];
+
+  const onThemeChange = (mode: string): void => {
+    setSelectedTheme(mode);
+  };
+
+  return (
+    <MuiAppBar position="static" color="inherit" variant="outlined" elevation={0} className={classes} {...rest}>
+      <Container maxWidth="xl">
+        <Toolbar disableGutters>
+          {isLeftNavigationActive && (
+            <IconButton aria-label="Menu Icon" onClick={onLeftNavigationTrigger} className="menu-icon-button">
+              <HamburgerIcon />
+            </IconButton>
+          )}
+          {brand && (
+            <Box
+              component={brand.onClick ? Link : Box}
+              className={clsx('logo-box', {
+                'with-link': Boolean(brand.onClick),
+              })}
+              onClick={brand.onClick}
+              underline="none"
+              color="inherit"
+            >
+              <Box className="logo">{isMobile ? brand.logo.mobile ?? brand.logo.desktop : brand.logo.desktop}</Box>
+              <Typography variant="h6" className="portal-name">
```

Comment:
avoid header tags for logos
use anchor
check brand related accessibility rules
Do a scan in lighthouse

Reply:
Tested for accessibility. Using MUI's default method of writing the brand name.